client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -687,71 +688,60 @@
+
+        function getCropCoordinates(cropName) {
+            var coordinates = {};
+            coordinates.CropLeft = Math.round(Math.min($scope.preview[cropName].cords.x, $scope.preview[cropName].cords.x2));
+            coordinates.CropRight = Math.round(Math.max($scope.preview[cropName].cords.x, $scope.preview[cropName].cords.x2));
+            coordinates.CropTop = Math.round(Math.min($scope.preview[cropName].cords.y, $scope.preview[cropName].cords.y2));
+            coordinates.CropBottom = Math.round(Math.max($scope.preview[cropName].cords.y, $scope.preview[cropName].cords.y2));
+
+            return coordinates;
+        }
 
-        function recordCrop(coordinates, cropName) {
-            $scope.data.renditions[cropName].CropLeft = coordinates.CropLeft;
-            $scope.data.renditions[cropName].CropTop = coordinates.CropTop;
-            $scope.data.renditions[cropName].CropRight = coordinates.CropRight;
-            $scope.data.renditions[cropName].CropBottom = coordinates.CropBottom;
+        function recordCrops(cropName) {
+            var obj = {};
+            obj[cropName] = getCropCoordinates(cropName);
+            _.extend($scope.data.cropData, obj);
         }
 
         $scope.done = function() {
-            var coordinates = {};
-            coordinates.CropLeft = Math.round(Math.min($scope.preview.cords.x, $scope.preview.cords.x2));
-            coordinates.CropRight = Math.round(Math.max($scope.preview.cords.x, $scope.preview.cords.x2));
-            coordinates.CropTop = Math.round(Math.min($scope.preview.cords.y, $scope.preview.cords.y2));
-            coordinates.CropBottom = Math.round(Math.max($scope.preview.cords.y, $scope.preview.cords.y2));
-
-            var cropName = $scope.data.cropsize.name;
-
-            //Save or overwrite crop if item is not published.
-            if ($scope.data.state !== 'published') {
-                return cropImage.saveCrop(coordinates, $scope.data).then(function(result) {
-                    var picture_url = result.renditions[cropName].href;
-                    notify.success(gettext('Image Cropped.'));
-                    $scope.data.picture_url = picture_url;
-                    recordCrop(coordinates, cropName);
-                }, function(response) {
-                    if (response.data._status === 'ERR'){
-                        notify.error(gettext('Error: ' + response.data._error.message));
-                    }
-                });
+            _.forEach($scope.data.cropsizes, function(cropsize) {
+                recordCrops(cropsize.name);
+            });
+            notify.success(gettext('Crop changes have been recorded'));
+            $scope.resolve($scope.data);
+        };
+
+        $scope.close = function() {
+            if ($scope.data.isDirty) {
+                modal.confirm(gettext('You have unsaved changes, do you want to continue?'))
+                .then(angular.bind(this, function record() { // Ok = continue w/o saving
```

Comment:
Seems like binding the function to `this` is redundant here? The `record()` function doesn't use `this` anywhere in its body...